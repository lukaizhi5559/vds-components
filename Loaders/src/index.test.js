import { render } from '@testing-library/react';
import React from 'react';
import { Loader } from '.';
import { ColorTokens } from '@vds-tokens/color';

function rgbToHex(rgb) {
  rgb = rgb
    .substr(4)
    .split(')')[0]
    .split(', ');

  let r = (+rgb[0]).toString(16),
    g = (+rgb[1]).toString(16),
    b = (+rgb[2]).toString(16);

  if (r.length == 1) r = '0' + r;
  if (g.length == 1) g = '0' + g;
  if (b.length == 1) b = '0' + b;

  return '#' + r + g + b;
}

describe('<Loader>', () => {
  test('<Loaders default Styles>', () => {
    const { container } = render(<Loader active />);
    //Loader - size and color
    const ld = container.firstChild.firstChild;
    expect(ld).toHaveStyleRule('height', '2.5rem');
    expect(ld).toHaveStyleRule('width', '2.5rem');
    expect(ld).toHaveStyleRule('border', '0.25rem solid #000000');
    //Behaviors - Animation
    expect(ld).toHaveStyleRule('animation', 'cVuWGa 0.5s infinite linear');
    //Background Overlay - color
    const overlay = container.firstChild;
    expect(overlay).toHaveStyleRule('background-color', '#ffffff');
    expect(overlay).toHaveStyleRule('opacity', '0.8');
  });

  test('<Loaders default Styles Inverted>', () => {
    const { container } = render(<Loader active surface="dark" />);
    //Loader - size and color
    const ld = container.firstChild.firstChild;
    expect(ld).toHaveStyleRule('height', '2.5rem');
    expect(ld).toHaveStyleRule('width', '2.5rem');
    expect(ld).toHaveStyleRule('border', '0.25rem solid #ffffff');
    //Behaviors - Animation
    expect(ld).toHaveStyleRule('animation', 'cVuWGa 0.5s infinite linear');
    //Background Overlay - color
    const overlay = container.firstChild;
    expect(overlay).toHaveStyleRule('background-color', '#000000');
    expect(overlay).toHaveStyleRule('opacity', '0.8');
  });

  test('should render Loader correctly', () => {
    const { container } = render(<Loader />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render active Loader correctly', () => {
    const { container } = render(<Loader active />);

    const loaderOverlay = container.firstChild;
    const styles = window.getComputedStyle(loaderOverlay)._values;
    expect(rgbToHex(styles['background-color'])).toBe(
      ColorTokens.background.primary.light.value
    );
    expect(loaderOverlay.getAttribute('aria-label')).toBe('loader overlay');

    const loader = loaderOverlay.firstChild;
    expect(loader.getAttribute('aria-label')).toBe('loader animation');
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render fullscren Loader correctly', () => {
    const { container } = render(<Loader fullscreen />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render active fullscreen Loader correctly', () => {
    const { container } = render(<Loader active fullscreen />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render inverted Loader correctly', () => {
    const { container } = render(<Loader surface="dark" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render active inverted Loader correctly', () => {
    const { container } = render(<Loader active surface="dark" />);

    const loaderOverlay = container.firstChild;
    const styles = window.getComputedStyle(loaderOverlay)._values;
    expect(rgbToHex(styles['background-color'])).toBe(
      ColorTokens.background.primary.dark.value
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should render active inverted fullscreen Loader correctly', () => {
    const { container } = render(<Loader active surface="dark" fullscreen />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
