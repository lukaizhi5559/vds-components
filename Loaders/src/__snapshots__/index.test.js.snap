// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader> should render Loader correctly 1`] = `null`;

exports[`<Loader> should render active Loader correctly 1`] = `
.c0 {
  background-color: #ffffff;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  z-index: 99999;
  position: absolute;
  display: block;
  -webkit-animation: eMLfYp 0.15s linear;
  animation: eMLfYp 0.15s linear;
  opacity: 0.8;
}

.c1 {
  outline: none;
  display: block;
  -webkit-animation: eMLfYp 0.15s linear;
  animation: eMLfYp 0.15s linear;
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  width: 2.5rem;
  height: 2.5rem;
  display: block;
  z-index: 3;
  border: 0.25rem solid #000000;
  border-top-color: transparent;
  border-radius: 50%;
  -webkit-animation: cVuWGa 0.5s infinite linear;
  animation: cVuWGa 0.5s infinite linear;
}

<div
  aria-label="loader overlay"
  class="c0"
>
  <div
    aria-label="loader animation"
    class="c1"
    tabindex="0"
  />
</div>
`;

exports[`<Loader> should render active fullscreen Loader correctly 1`] = `null`;

exports[`<Loader> should render active inverted Loader correctly 1`] = `
.c0 {
  background-color: #000000;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  height: 100%;
  z-index: 99999;
  position: absolute;
  display: block;
  -webkit-animation: eMLfYp 0.15s linear;
  animation: eMLfYp 0.15s linear;
  opacity: 0.8;
}

.c1 {
  outline: none;
  display: block;
  -webkit-animation: eMLfYp 0.15s linear;
  animation: eMLfYp 0.15s linear;
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: auto;
  width: 2.5rem;
  height: 2.5rem;
  display: block;
  z-index: 3;
  border: 0.25rem solid #ffffff;
  border-top-color: transparent;
  border-radius: 50%;
  -webkit-animation: cVuWGa 0.5s infinite linear;
  animation: cVuWGa 0.5s infinite linear;
}

<div
  aria-label="loader overlay"
  class="c0"
>
  <div
    aria-label="loader animation"
    class="c1"
    tabindex="0"
  />
</div>
`;

exports[`<Loader> should render active inverted fullscreen Loader correctly 1`] = `null`;

exports[`<Loader> should render fullscren Loader correctly 1`] = `null`;

exports[`<Loader> should render inverted Loader correctly 1`] = `null`;
